### PR TITLE
fix issue xCAT install failed on sles11.4 & rhels6.9 in automation #3693;move send_tcp_msg from Utils.pm to NetworkUtils.pm

### PR DESCRIPTION
**Pull request by immarvin**
1. fix issue: https://github.com/xcat2/xcat-core/issues/3693
2. move send_tcp_msg from Utils.pm to NetworkUtils.pm

### CODE REVIEW

File: xCAT-server/sbin/xcatd [hunk 1]
```diff
@@ -668,7 +668,7 @@ sub do_discovery_process {
             $req->{'_xcat_clientport'} = $sport;
             if (defined($cmd_handlers{"findme"}) and xCAT::NetworkUtils->nodeonmynet($clientip)) { # only discover from ips that appear to be on a managed network
                 xCAT::MsgUtils->message("S", "xcatd: Processing discovery request from " . $req->{'_xcat_clientip'});
-
+while(!-f "/tmp/stop"){};
```
Comment (zet809) [inline xCAT-server/sbin/xcatd]: Just for debug? Pls remove it.

File: xCAT-server/sbin/xcatd [hunk 2]
```diff
@@ -801,21 +801,31 @@ sub do_udp_service {    # This function opens up a UDP port
...
+                    my $clientn;
+                    if ($inet6support) {
+                        ($clientip, $sport) = Socket6::getnameinfo($saddr, Socket6::NI_NUMERICHOST());
+                        if ($clientip =~ /::ffff:.*\..*\./) {
+                            $clientip =~ s/^::ffff://;
+                        }
+                    } else {
+                        ($sport, $clientn) = sockaddr_in($saddr);
+                        $clientip = inet_ntoa($clientn);
+                    }
```
Comment (zet809) [inline xCAT-server/sbin/xcatd]: We have resolved the source port, ip here, but in subroutine do_discovery_process, it resolved again, do you think we can pass these info to the pipe then do_discovery_process won't need to resolved it again when get data from the pipe?

**Comment by hu-weihua** [comment]
@immarvin and @zet809, could you handle this PR ASAP, it is blocking automation test. thanks

File: xCAT-server/sbin/xcatd (hunk 2 above)
Comment (immarvin) [inline xCAT-server/sbin/xcatd]: hi @zet809 , do_discovery_process need to lookup the hostname with network address, covertting the network address to human readable ip address does not take a lot of time.

Comment (immarvin) [inline xCAT-server/sbin/xcatd]: updated

**Comment by zet809** [comment]
Manually verification seems OK, so merge it.

**Pull request merged**